server/rating_service/rating_service.py

```
@@ -0,0 +1,254 @@
+    pass
+
+
+class ServiceNotReadyError(RatingServiceError):
+    pass
+
+
+class RatingNotFoundError(RatingServiceError):
+    pass
+
+
+@with_logger
+class RatingService:
+    """
+    Service responsible for calculating and saving trueskill rating updates.
+    To avoid race conditions, rating updates from a single game ought to be
+    atomic.
+    """
+
+    def __init__(self, database: FAFDatabase, player_service: PlayerService):
+        self._db = database
+        self._player_service_callback = player_service.signal_player_rating_change
+        self._accept_input = False
+        self._queue = asyncio.Queue()
+        self._task = None
+
+    async def initialize(self):
+        # TODO use create_task in python3.7
+        self._accept_input = True
+        self._logger.debug("RatingService starting...")
+        self._task = asyncio.ensure_future(self._handle_rating_queue())
+
+    async def enqueue(self, summary: GameRatingSummary) -> None:
+        if not self._accept_input:
+            self._logger.warning("Dropped rating request %s", summary)
+            raise ServiceNotReadyError(
+                "RatingService not yet initialized or shutting down."
+            )
+
+        self._logger.debug("Queued up rating request %s", summary)
+        await self._queue.put(summary)
+        rating_service_backlog.set(self._queue.qsize())
+
+    async def _handle_rating_queue(self):
+        self._logger.info("RatingService started!")
+        while True:
+            summary = await self._queue.get()
+            self._logger.debug("Now rating request %s", summary)
+
+            try:
+                await self._rate(summary)
+            except GameRatingError:
+                self._logger.warning("Error rating game %s", summary)
+            except RatingNotFoundError:
+                self._logger.warning("Missing rating entry to rate game %s", summary)
+            else:
+                self._logger.debug("Done rating request.")
```

If `self._rate` throws any other error that isn't caught, then the loop ends, the task exits, and we won't rate any more games while the input queue still keeps accepting input.

Maybe this is the place to add a bare except, log the error, and move on. Is that really the cleanest solution?